universal-login-commons/test/core/utils/generateBackupCodes.ts

```diff
@@ -0,0 +1,117 @@
+import {expect} from 'chai';
+import {toWords, generateBackupCode, fromWords} from '../../../lib/core/utils/generateBackupCodes';
+import {utils} from 'ethers';
+
+describe('Daefen', () => {
+  describe('toWords', () => {
+    it('Single syllable', () => {
+      expect(toWords(utils.bigNumberify(0))).to.eq('Ab');
+      expect(toWords(utils.bigNumberify(1))).to.eq('Ac');
+      expect(toWords(utils.bigNumberify(1000))).to.eq('Cen');
+      expect(toWords(utils.bigNumberify(3455))).to.eq('Yzy');
+    });
+
+    it('Two syllables', () => {
+      expect(toWords(utils.bigNumberify(3456))).to.eq('Acab');
+      expect(toWords(utils.bigNumberify(3457))).to.eq('Acac');
+      expect(toWords(utils.bigNumberify(6912))).to.eq('Adab');
+      expect(toWords(utils.bigNumberify(11943935))).to.eq('Yzyyzy');
+    });
+
+    it('More syllables', () => {
+      expect(toWords(utils.bigNumberify(11943936))).to.eq('Acab Ab');
+    });
+
+    it('More syllables', () => {
+      expect(toWords(utils.bigNumberify(11943936))).to.eq('Acab Ab');
+    });
+
+    it('Max allowed number in input 2^53-1 passed as number', () => {
+      expect(toWords(utils.bigNumberify(9007199254740991))).to.eq('Omkeu Ugekum Ope');
+    });
+
+    it('Error for 2^53 passed as number', () => {
```

Comment:
Why? Should support all the way to 2^256.

Reply:
All number under 2^53 should be passed as a string. For example: `'100000'` instead of `100000`